### DISCUSSION TITLE
whether to consider cache vertexDataFrame

### DISCUSSION THREAD

**Discussion by haohao0103** [apache/incubator-graphar#256] [General]
The vertexDataFrame obtained when the VertexWriter is created is convenient for subsequent jobs to use, whether to consider cache;
com.alibaba.graphar.writer.VertexWriter.scala
![image](https://github.com/alibaba/GraphAr/assets/30570711/df9a8564-1df9-47cc-98be-88b1d93f4aba)

**Comment by acezen** [comment]
It's maybe a good way to cache the DataFrame, but in writing process, the DataFrame is always been used only once, I not sure shall we need to cache it.

**Reply by haohao0103** [reply]
hi, thank you for your reply, Caching RDDS may seem unreasonable when we first created the VertextWriter, I think it makes more sense to cache chunks when creating the VertextWriter, because creating chunks is a shuffle operation and writing them out for each propertyGroup, which can happen multiple times. This is my understanding. I look forward to your reply.
![image](https://github.com/alibaba/GraphAr/assets/30570711/76748d09-77bf-4224-8f39-f486eb8a36e2)

**Reply by acezen** [reply]
sorry for reply this so late. It seems that cache the chunks is a good optimization for memory performance. And I think there are some other place that can apply the same optimization too. Can you help us to improve GraphAr with the optimization? look forward to your reply.

**Comment by haohao0103** [comment]
&nbsp;Thanks, I am glad to do that.




没脑子?
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "alibaba/GraphAr"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月12日(星期五) 晚上6:04
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [alibaba/GraphAr] whether to consider cache vertexDataFrame (Discussion #256)





 
sorry for reply this so late. It seems that cache the chunks is a good optimization for memory performance. And I think there are some other place that can apply the same optimization too. Can you help us to improve GraphAr with the optimization? look forward to your reply.

**Reply by acezen** [reply]
The issue #322 has been created for this discussion, FYI.
